sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/CodeGenInternal/PatchModels/WebhookHookParameterPatch.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.AI.MetricsAdvisor.Models
+{
+    internal partial class WebhookHookParameterPatch
+    {
+        public IDictionary<string, string> Headers { get; internal set; }
+
+        // Full qualification must be used so CodeGen skips the generation of this method.
+        // See: https://github.com/Azure/autorest.csharp/issues/793
+        void global::Azure.Core.IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
+        {
+            writer.WriteStartObject();
+            writer.WriteNullStringValue("endpoint", Endpoint);
+            writer.WriteNullStringValue("username", Username);
+            writer.WriteNullStringValue("password", Password);
```

Comment:
can this be null? same with the certificates parameters

Reply:
That's a good question, but yes. The service does not return secrets for data sources, but it does for hooks. The secrets used in hooks do not grant access user data, though, so I can understand them being less cautious with it.

A webhook is an endpoint created and provided by the customer. The password and certificate here are used to let the service send notifications to the customer-provided endpoint, but no user data is sent back.